File: Algorithms/src/1861. Rotating the Box/Code/solution.go
func rotateTheBox(boxGrid [][]byte) [][]byte {

    m := len(boxGrid)
    n := len(boxGrid[0])

    // Process every row
    for row := 0; row < m; row++ {

        // Rightmost empty position
        emptyCol := n - 1

        // Traverse from right to left
        for col := n - 1; col >= 0; col-- {

            // Obstacle found
            if boxGrid[row][col] == '*' {

                // Reset valid position
                emptyCol = col - 1

            // Stone found
            } else if boxGrid[row][col] == '#' {

                // Remove current stone
                boxGrid[row][col] = '.'

                // Move stone
                boxGrid[row][emptyCol] = '#'

                // Update next empty position
                emptyCol--
            }
        }
    }

    // Create rotated matrix
    rotated := make([][]byte, n)

    for i := 0; i < n; i++ {
        rotated[i] = make([]byte, m)
    }

    // Rotate clockwise
    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {

            rotated[j][m-1-i] = boxGrid[i][j]
        }
    }

    return rotated
}

File: Algorithms/src/1861. Rotating the Box/Code/solution.js
/**
 * @param {character[][]} boxGrid
 * @return {character[][]}
 */
var rotateTheBox = function (boxGrid) {
  const m = boxGrid.length;
  const n = boxGrid[0].length;

  // Process each row
  for (let row = 0; row < m; row++) {
    // Rightmost valid empty position
    let emptyCol = n - 1;

    // Traverse from right to left
    for (let col = n - 1; col >= 0; col--) {
      // Obstacle found
      if (boxGrid[row][col] === "*") {
        // Reset empty position
        emptyCol = col - 1;
      }

      // Stone found
      else if (boxGrid[row][col] === "#") {
        // Remove current stone
        boxGrid[row][col] = ".";

        // Move stone
        boxGrid[row][emptyCol] = "#";

        // Update next empty spot
        emptyCol--;
      }
    }
  }

  // Create rotated matrix
  const rotated = Array.from({ length: n }, () => Array(m).fill("."));

  // Rotate clockwise
  for (let i = 0; i < m; i++) {
    for (let j = 0; j < n; j++) {
      rotated[j][m - 1 - i] = boxGrid[i][j];
    }
  }

  return rotated;
};
